core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -396,9 +395,23 @@ abstract class AbstractControllerBrokerRequestBatch(config: KafkaConfig,
   def addStopReplicaRequestForBrokers(brokerIds: Seq[Int],
                                       topicPartition: TopicPartition,
                                       deletePartition: Boolean): Unit = {
+    // A sentinel (-2) is used as an epoch if the topic is queued for deletion or
+    // does not have a leader yet. This sentinel overrides any existing epoch.
+    val leaderEpoch = if (controllerContext.isTopicQueuedUpForDeletion(topicPartition.topic)) {
+      LeaderAndIsr.EpochDuringDelete
+    } else {
+      controllerContext.partitionLeadershipInfo.get(topicPartition)
+        .map(_.leaderAndIsr.leaderEpoch)
+        .getOrElse(LeaderAndIsr.EpochDuringDelete)
+    }
+
     brokerIds.filter(_ >= 0).foreach { brokerId =>
```

Comment:
No need to fix here, but do you know why we do this filtering?

Reply:
I've asked myself the same question but I couldn't find a reason. I believe that `brokerId` is always >= 0 in the controller.